### PR TITLE
Updates to fip-0086.md 

### PR DESCRIPTION
**Pull request by ranchalp**
- Implemented changes to synchronization of participants (only exponential back-off atm)
- Mention to heaviest tipset not being actually about weight but blockheight suffices (clarification).

### CODE REVIEW

**Comment by ranchalp** [comment]
@jsoares @anorth see.

Remaining changes (TODO FYI, *not* made in this PR): 
- MakeTicket including instance ID (and some other calls of pseudocode?) (See [go-f3#68](https://github.com/filecoin-project/go-f3/issues/68))
- Optimization for the first round (See [go-f3#95](https://github.com/filecoin-project/go-f3/issues/95))

**Review by anorth — APPROVED** [review]

**Review by jsoares — APPROVED** [review]
Changes look good. We could incorporate the couple of todos in this same PR, but fine to do it separately too.

**Comment by ranchalp** [comment]
I will make them here directly and flag when ready.

**Comment by ranchalp** [comment]
@jsoares @anorth the changes I listed above are done now (and a couple typos corrected). 
Potential TODOs (asking for your opinion). I may do a full review of the FIP in time for the audit, but that can come in a different PR to be merged in time for the audit if (i) I manage to do it and (ii) there are important changes that matter for the audit, which I do not expect.

I also noticed the following TODO in the FIP: 

- [ ] Update and move proofs of correctness and other supplemental documentation into `resources`

The only change I can think of that involves the proofs could be the 2-phase optimization, or we could have a separate section on why the 2-phase optimization does not change anything. But I think this is not a priority for the audit (lmk otherwise and I will find time for this to be done timely).

Otherwise, ready to be merged and we can give closure to FIP changes pre-audit.

**Review by jsoares — APPROVED** [review]

**Comment by jsoares** [comment]
> Potential TODOs (asking for your opinion). I may do a full review of the FIP in time for the audit, but that can come in a different PR to be merged in time for the audit if (i) I manage to do it and (ii) there are important changes that matter for the audit, which I do not expect.

I'm actually getting a bit nervous about spec and implementation diverging but I trust you on this as you've been knee-deep in both. 

>  Update and move proofs of correctness and other supplemental documentation into resources

I think we'll probably want to review the proofs in time for the internal audit -- or at least clearly identify any deltas. It needn't happen by Monday. 

I'm okay with merging as is, pending @anorth's re-review (@anorth : please do it by EOW so that we leave this in a good state for the audit 🙏 ).

**Comment by ranchalp** [comment]
anorth already approved the request, but merging is blocked until the rest also approve.

> I'm actually getting a bit nervous about spec and implementation diverging but I trust you on this as you've been knee-deep in both.

[At the time of writing multi-instance consensus is not yet merged nor finished in the code] Actually, the code is quite in line with the spec as detailed in this PR, so need not worry about it I think. But my availability for discussions with auditors could solve whatever we may be missing.

**Review by jennijuju — APPROVED** [review]
Editorial: LGTM

**Review by anorth — CHANGES_REQUESTED** [review]
This has changed significantly since I reviewed it. Please don't do that 😢 . Retracting my approval until I have time to think about the new changes here. You can get approval back quickly by reverting to only trivial changes.

File: FIPS/fip-0086.md [hunk 1]
```diff
@@ -482,12 +494,10 @@ GossiPBFT ensures termination provided that (i) all participants start the insta
...
 
-The synchronization of participants is performed in the call to updateTimeout(timeout, round) (line 47), and works as follows:
+The synchronization of participants is performed in the call to updateTimeout(timeout, round) (line 52), and works as follows:
 
 * Participants start an instance with $Δ=2s$.
-* If the first and second rounds fail to reach termination, participants start the 3rd round with $Δ=3s$.
-* If 5 rounds fail to reach termination (rounds from 0 to 4), participants wait to receive the next drand epoch value to start round 5. This ensures that a potential ε delay between when participants started the instance is not carried over beyond round 4.
-* If round 5 fails to reach a decision, participants set exponential timeout increase $Δ= 1.3*Δ$ for every failed round. This ensures termination even if the real network delay increases arbitrarily.
+* Participants set their timeout for the current round to $2*1.3^{r}$ where $r$ is the round number ($r=0$ for the first round).
```
Comment (anorth) [inline FIPS/fip-0086.md:500]:
```suggestion
* Participants set their timeout for the current round to $Δ*1.3^{r}$ where $r$ is the round number ($r=0$ for the first round).
```

**Review by anorth — COMMENTED** [review]
As technical review, I am not confident that the optimisation of skipping a phase is a good one, because I don't believe the assumptions. More https://github.com/filecoin-project/go-f3/issues/95#issuecomment-1989654998

Everything else looks ok.

**Comment by jsoares** [comment]
@anorth I think I managed to revert the 2-phase optimisation changes without breaking anything else. Can you review & merge if OK? We should really get this in today.

File: FIPS/fip-0086.md [hunk 2]
```diff
@@ -431,6 +433,7 @@ Valid(m):                                   | For a message m to be valid,
 
 If m is not properly signed:                | m must be properly signed.
   return False
+if m.instance != instance                   | m must match the instance ID
```
Comment (anorth) [inline FIPS/fip-0086.md:436]: I think this should have `return False` below it.

**Review by anorth — APPROVED** [review]

File: FIPS/fip-0086.md (hunk 2 above)
Comment (jsoares) [reply FIPS/fip-0086.md:436]:
```suggestion
if m.instance != instance                   | m must match the instance ID
  return False
```

Seems to have been missing already in the original commit, but indeed.

**Pull request merged**